src/Context/cartContext.js

```diff
@@ -0,0 +1,43 @@
+import { createContext, useContext, useReducer } from "react";
+
+const CartContext = createContext();
+
+const cartFunction = (cartState, action) =>{
+    switch(action.type) {
+        case 'Add_to_cart':
+            return {...cartState,
+               cart: [
+                  ...cartState.cart, 
+                  {...action.payload, quantity: 1}]}
+
+                  case 'Increase_quantity':
+                    return {...cartState,
+                       cart: 
+                        cartState.cart.map((item) => item.id === action.payload.id ? 
+                            {...item, quantity: item.quantity + 1} : item)
```

Comment:
I think you dont need to `increase` or `decrease` explicitly . you can use backend for this .

Reply:
Oh okay, will try to change that.